pkg/controller/postgresql/role/reconciler.go

```diff
@@ -148,10 +147,25 @@ func privilegesToClauses(p v1alpha1.RolePrivilege) []string {
 	negateClause("REPLICATION", p.Replication, &pc)
 	negateClause("BYPASSRLS", p.BypassRls, &pc)
 
-	sort.Strings(pc)
 	return pc
 }
 
+func changedPrivs(existing []string, desired []string) []string {
+	out := []string{}
+
+	// The input slices here are outputted by privilegesToClauses above.
+	// Because these are created by repeated calls to negateClause in the
+	// same order, we can rely on each clause being in the same array
+	// position in the 'desired' and 'existing' inputs.
+
+	for i, v := range desired {
+		if v != existing[i] {
```

Comment:
Could the slice of existing privs be shorter than the slice of desired privs, causing a panic here?

Reply:
Both lists are produced by `privilegesToClauses()` which will always create arrays of the same length and order. If a clause was added in a later version of the provider, then the list of privileges in `status.atProvider.PrivilegesAsClauses` would in theory be shorter.

I think that case is covered on upgrade because `Observe()` would always run first, so the correct length value should always be set, however - I suppose you could still edit the status field manually and cause a panic.

I'll add a length comparison just in case and error out 👍